cpp/src/container_test.cpp

```diff
@@ -179,6 +181,22 @@ int test_container_capabilities() {
     ASSERT_EQUAL(th.peer_offered_capabilities[0], proton::symbol("offered"));
     ASSERT_EQUAL(th.peer_desired_capabilities.size(), 1u);
     ASSERT_EQUAL(th.peer_desired_capabilities[0], proton::symbol("desired"));
+    ASSERT_EQUAL(th.peer_properties.size(), 0u);
+    return 0;
+}
+
+int test_container_properties() {
+    proton::connection_options opts;
+    std::map<proton::symbol, proton::value> props = {
+            {"qpid.client_process", "test_process"},
+            {"qpid.client_pid", 123}
+    };
```

Comment:
Irritatingly we still support C++03 (for a little while longer only I think) so this syntax is not acceptable just yet.

Reply:
Do we? @ssorj afaik we don't have to support C++03 any more on what's to become Proton 0.34, or am I wrong?

Reply:
At this point we still support C++03. I think 0.34 will be the last release to do this (and also python 2.7) We should announce a one release deprecation on the mailing list, and have a clean break from Proton 0.34 to 0.35.

Reply:
True, nothing was announced yet, that's a fact. Alright, so 03 is still supported.

Reply:
Thanks for pointing this out, I will change it to something C++03 conformant. 
If C++03 is still being supported, might it be beneficial to have a test build that explicitly uses that variant?

Reply:
Yep, it would make a lot of sense to have an C++03 build. However given it really is going away in a short while probably not worth the bother right now - sorry you had to fix this for 03 when it's going away soon!

Reply:
I agree, a bit late in the day.  However, I found the required setting for cmake to build with C++03 - which fortunately showed up another compatibility error in the PR.